### PR TITLE
libslax: fix build for Linux

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/libslax.rb [hunk 1]
```diff
@@ -20,27 +21,53 @@ class Libslax < Formula
...
-    ENV.append "LDFLAGS", "-L#{Formula["openssl@1.1"].opt_lib}"
+    # Upstream patches have already bumped package version in configure.ac ahead of new release being tagged;
+    # remove on next release
+    inreplace "configure.ac",
+              "AC_INIT([libslax],[0.22.2],[phil@juniper.net])",
+              "AC_INIT([libslax],[#{version}],[phil@juniper.net])"
+
+    # Upstream setup.sh script will not overwrite/replace an existing configure script
+    rm "configure"
+    system "sh", "./bin/setup.sh"
```
Comment (danielnachun) [inline Formula/libslax.rb:65]: Could `autoreconf --force --verbose --install` work here instead?  It looks like the `setup.sh` script just runs `autoreconf --install` which won't overwrite the old `configure` script as you've noted.

File: Formula/libslax.rb [hunk 2]
```diff
@@ -20,27 +21,51 @@ class Libslax < Formula
...
+    # Upstream patches have already bumped package version in configure.ac ahead of new release being tagged,
+    # remove on next release
+    inreplace "configure.ac",
+              "AC_INIT([libslax],[0.22.2],[phil@juniper.net])",
+              "AC_INIT([libslax],[#{version}],[phil@juniper.net])"
+
+    system "autoreconf", "--force", "--install", "--verbose"
+
+    args = std_configure_args + %w[--enable-libedit]
+    args << "--with-sqlite3=#{Formula["sqlite"].opt_prefix}" unless OS.linux?
```
Comment (danielnachun) [inline Formula/libslax.rb:66]: Do we know if this actually requires using brewed `sqlite` on macOS?  If so we should probably add a comment explaining why.

Comment (alebcay) [reply Formula/libslax.rb:66]: I think I actually mixed this up, it should be `if OS.linux?`. The `sqlite` detection works fine as-is on macOS (using system `sqlite`).

**Review by danielnachun — APPROVED** [review]

**Comment by alebcay** [comment]
https://github.com/Homebrew/homebrew-core/actions/runs/2553288819

**Pull request closed without merging**